study1/week1/Third Maximum Number.java

```diff
@@ -0,0 +1,30 @@
+class Third Maximum Number {
+
+    int count = 3;            
+    int tmp = Integer.MIN_VALUE;
+
+    public int thirdMax(int[] nums) {              // Space Complexity= O(n), n is the length of the Array
+      Arrays.sort(nums);                          // time complexity = O(n * log(n))
+      if(nums.length <3){    
+          return nums[nums.length-1];
+      }else{
+         for(int i=nums.length-1; i>=0 ; i--){   // time complexity = O(m), m is the position, where the third maximum number is set
```

Comment:
you cannot consider `m` as a parameter, as it's totally random, and a thing that does not depend on the input size.